test/snippet/io/sequence_file/sequence_file_output_fields_trait_2.cpp

```diff
@@ -0,0 +1,32 @@
+#include <sstream>
+
+#include <seqan3/io/sequence_file/input.hpp>
+#include <seqan3/io/sequence_file/output.hpp>
+
+auto input = R"(@TEST1
+ACGT
++
+##!#
+@Test2
+AGGCTGA
++
+##!#!!!
+@Test3
+GGAGTATAATATATATATATATAT
++
+##!###!###!###!###!###!#)";
+
+int main()
+{
+    seqan3::sequence_file_input  fin{std::istringstream{input},
+                                     seqan3::format_fastq{},
+                                     seqan3::fields<seqan3::field::SEQ, seqan3::field::ID, seqan3::field::QUAL>{}};
+    seqan3::sequence_file_output fout{std::ostringstream{},
+                                      seqan3::format_fastq{}}; // doesn't have to match the configuration
+
+    for (auto & r : fin)
+    {
+        if (true) // r fulfills some criterium
```

Comment:
I don't understand this snippet. What is it trying to explain?

Reply:
Here's the comment before the snippet: 

 ```cpp* A different way of passing custom fields to the file is to pass a seqan3::record – instead of a tuple – to
 * push_back(). The seqan3::record clearly indicates which of its elements has which seqan3::field ID so the file will
 * use that information instead of the template argument. This is especially handy when reading from one file and
 * writing to another, because you don't have to configure the output file to match the input file, it will just work:```